File: VB/FillEmptySpaceGridColumn/Form1.vb
Imports System
Imports System.Windows.Forms

Namespace FillEmptySpaceGridColumn
    Partial Public Class Form1
        Inherits Form

        Public Sub New()
            InitializeComponent()
        End Sub

        Private Sub Form1_Load(ByVal sender As Object, ByVal e As EventArgs) Handles MyBase.Load
            Me.carsTableAdapter.Fill(Me.carsDBDataSet_Renamed.Cars)
        End Sub

        Private Sub simpleButton1_Click(ByVal sender As Object, ByVal e As EventArgs) Handles simpleButton1.Click
            fillGridView1.Columns("HP").Visible = Not fillGridView1.Columns("HP").Visible
        End Sub
    End Class

End Namespace

File: CS/FillEmptySpaceGridColumn/Form1.cs
using System;
using System.Windows.Forms;

namespace FillEmptySpaceGridColumn
{
	public partial class Form1 : Form
	{
		public Form1()
		{
			InitializeComponent();
		}

		private void Form1_Load(object sender, EventArgs e)
		{
			this.carsTableAdapter.Fill(this.carsDBDataSet.Cars);
		}

        private void simpleButton1_Click(object sender, EventArgs e)
        {
            fillGridView1.Columns["HP"].Visible = !fillGridView1.Columns["HP"].Visible;
        }
	}

}